mixins/segment-mixin.ts

```diff
@@ -0,0 +1,57 @@
+declare global {
+  interface Window {
+    digitalData: any
+    _analytics: any,
+    bluemixAnalytics: any
+  }
+}
+
+export const segmentMixin = {
+  created () {
+    const self = this as any
+
+    if (process.client) {
+      window.digitalData = {
+        page: {
+          pageInfo: {
+            productTitle: self.title,
+            analytics: {
+              category: 'Qiskit.org'
+            }
+          }
+        }
+      }
+
+      window._analytics = {
+        segment_key: 'ffdYLviQze3kzomaINXNk6NwpY9LlXcw',
+        coremetrics: false,
+        optimizely: false,
+        googleAddServices: false,
+        fullStory: false,
+        autoPageEventSpa: false,
+        autoFormEvents: false,
+        autoPageView: false
+      }
+    }
+  },
+  beforeRouteEnter (to, from, next) {
+    next((self) => {
+      const routeName: string = self.belongsTo
+      const title: string = self.$metaInfo.title
+      const path: string = to.path
+
+      if (window.bluemixAnalytics && window.bluemixAnalytics.pageEvent) {
```

Comment:
I'm not sure this would also take care of the first load? (like would this fire up when the first page from the server renders or only after a route change on the client?)

Reply:
I've just checked it and you're right, it doesn't take care of the first load. I will fix it, thank you!